components/focus-trap/focus-trap.js

```diff
@@ -68,6 +70,17 @@ class FocusTrap extends FocusMixin(LitElement) {
 		`;
 	}
 
+	willUpdate(changedProperties) {
+		if (!changedProperties.has('trap')) return;
+		if (this.trap) {
+			traps.push(this);
+		} else {
+			const trapIndex = traps.findIndex(trap => trap === this);
+			if (trapIndex === -1) return;
+			traps.splice(trapIndex, 1);
```

Comment:
A `pop()` could be used here instead, but I felt the look-up and `splice()` was safer in case multiple dialogs are closed at once and their traps are removed in an order that doesn't match our `traps` array.